Snakefile

```diff
@@ -583,6 +630,68 @@ rule host_filter_pe:
                   """)
 
 
+rule bloom_filter_pe:
+    """
+    Performs bloom read filtering on paired end data using Bowtie and Samtools/
+    BEDtools. Takes the four output files generated after host filtering. 
+
+    Also requires an indexed reference (path specified in config). 
+
+    First, uses Bowtie output piped through Samtools to only retain read pairs
+    that are never mapped (either concordantly or just singly) to the indexed
+    reference genome. Fastqs from this are gzipped into matched forward and 
+    reverse pairs. 
+
+    Unpaired forward and reverse reads are simply run through Bowtie and
+    non-mapping gzipped reads output.
+
+    All piped output first written to localscratch to avoid tying up filesystem.
+    """
+    input:
+        forward  = "data/{sample}/{run}/host_filtered/{sample}_R1.trimmed.host_filtered.fq.gz",
+        reverse  = "data/{sample}/{run}/host_filtered/{sample}_R2.trimmed.host_filtered.fq.gz",
+        unpaired_1 = "data/{sample}/{run}/host_filtered/{sample}_U1.trimmed.host_filtered.fq.gz",
+        unpaired_2 = "data/{sample}/{run}/host_filtered/{sample}_U2.trimmed.host_filtered.fq.gz"
+    output:
+        forward  = "data/{sample}/{run}/bloom_filtered/{sample}_R1.trimmed.bloom_filtered.fq.gz",
+        reverse  = "data/{sample}/{run}/bloom_filtered/{sample}_R2.trimmed.bloom_filtered.fq.gz",
+        unpaired_1 = "data/{sample}/{run}/bloom_filtered/{sample}_U1.trimmed.bloom_filtered.fq.gz",
+        unpaired_2 = "data/{sample}/{run}/bloom_filtered/{sample}_U2.trimmed.bloom_filtered.fq.gz"
+    params:
+        forward_fn = "{sample}_R1.trimmed.bloom_filtered.fq",
+        reverse_fn = "{sample}_R2.trimmed.bloom_filtered.fq",
+        unpaired_1_fn = "{sample}_U1.trimmed.bloom_filtered.fq.gz",
+        unpaired_2_fn = "{sample}_U2.trimmed.bloom_filtered.fq.gz",
+        bloom_db = config["BLOOM_DB"]
+    threads:
+        12
+    benchmark:
+        "benchmarks/{run}/qc/bloom_filter_pe_{sample}.json"
+    log:
+        bowtie = "logs/{run}/bowtie/{sample}.log",
+        other = "logs/{run}/qc/bloom_filter_pe_{sample}.log" 
+    run:
+        with tempfile.TemporaryDirectory(dir=TMP_DIR_ROOT) as temp_dir:
+            shell("""
+                  set +u; {BOWTIE_ENV}; set -u
+
+                  bowtie2 -p {threads} -x {params.bloom_db} --very-sensitive -1 {input.forward} -2 {input.reverse} 2> {log.bowtie}| \
```

Comment:
I have a hunch we will want to change this to leave in discordantly mapped reads in future versions...